src/Dapr.Client/DaprClient.cs

```diff
@@ -939,6 +940,293 @@ public async Task<StateEntry<TValue>> GetStateEntryAsync<TValue>(string storeNam
+        public abstract Task<byte[]> EncryptAsync(string vaultResourceName, byte[] plainTextBytes,
+            KeyWrapAlgorithm algorithm, string keyName, DataEncryptionCipher dataEncryptionCipher = DataEncryptionCipher.AesGcm,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Encrypts an array of bytes using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="plainTextBytes">The bytes of the plaintext value to encrypt.</param>
+        /// <param name="algorithm">The name of the algorithm used to wrap the encryption key.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+        /// <param name="decryptionKeyName">The name (and optionally version) of the decryption key to specify should be used.</param>
+        /// <param name="dataEncryptionCipher">The name of the cipher to use for the encryption operation.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<byte[]> EncryptAsync(string vaultResourceName, byte[] plainTextBytes,
+            KeyWrapAlgorithm algorithm, string keyName, string decryptionKeyName, DataEncryptionCipher dataEncryptionCipher = DataEncryptionCipher.AesGcm,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Encrypts an array of bytes using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="plainTextStream">The stream containing the bytes of the plaintext value to encrypt.</param>
+        /// <param name="algorithm">The name of the algorithm used to wrap the encryption key.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+        /// <param name="dataEncryptionCipher">The name of the cipher to use for the encryption operation.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<byte[]> EncryptAsync(string vaultResourceName, Stream plainTextStream,
```

Comment:
I wonder if it'd make sense to offer a streamed response (e.g. based on `IAsyncEnumerable`) if provided an input stream?

Reply:
I mentioned this in one of the [comments](https://github.com/dapr/dotnet-sdk/blob/0e92eae547e5800e5501732df65f6a525a99e1d9/src/Dapr.Client/DaprClientGrpc.cs#L1519), but the only reason I didn't add one here is because of the [conversation](https://discord.com/channels/778680217417809931/1133104673558114439/1187858033091563591) I had with @ItalyPaleAle in the Crypto channel on Discord in which they said that "gRPC doesn't allow continuous streams of data" which clarified that the data being encrypted is bounded. As such, as part of the implementation in DaprClientGrpc, I read this back out to the byte array and return it.

The idea was that by only exposing methods with a bounded length, we don't encourage developers to build something that would enable unbounded streams since those aren't actually supported at the sidecar level (though it's not obvious from the .NET SDK itself). However, if you think that's something better addressed through documentation, I'm not otherwise opposed (especially since I don't know the background of this limitation - if it's something being worked on, perhaps this overload should be added when that work is completed).

Reply:
I think there are plenty of uses in .NET of APIs that make use of values that are *potentially* unbounded (think `IEnumerable`) but which are generally expected to be bounded, and there generally isn't a problem.  Good, explicit documentation can help there.

Reply:
I'm sold - updated the implementation last night accordingly